### PR TITLE
Create CONTRIBUTING.md and UPGRADING.md

### PR DESCRIPTION
**Pull request by NitinAgg**
# Description

# Safety checklist
* [ ] This change is backwards compatible and safe to apply by existing users
* [ ] This change will NOT lead to data loss
* [ ] This change will NOT lead to downtime who already has an env/service setup

## How has this change been tested, beside unit tests?
YOUR_ANSWER

### CODE REVIEW

File: CONTRIBUTING.md
Comment (bigbitbus) [inline CONTRIBUTING.md:67]: Contributors may not know how to test this - whether their PR is breaking the main branch. We should point them to which tests they need to perform before submitting PR (e.g. unit tests, linting with scripts/lint.py etc) , and then which ones will be performed by the core team later (e.g. CircleCI).

Comment (NitinAgg) [reply CONTRIBUTING.md:67]: So we already have a setting that if unit tests etc are failing the PR cannot be merged. I think it's fine to just have this line as a guideline. If someone still sends a PR with broken tests, we can remind them of this. They can't merge it anyway.

**Review by juandiegopalomino — APPROVED** [review]

File: CONTRIBUTING.md
Comment (RemyDeWolf) [inline CONTRIBUTING.md:27]: all this information and some boiler plate can be in a issue template
ex: https://github.com/DataDog/datadog-agent/blob/main/.github/ISSUE_TEMPLATE.md

File: CONTRIBUTING.md
Comment (RemyDeWolf) [inline CONTRIBUTING.md:41]: Tag -> Label

File: CONTRIBUTING.md
Comment (RemyDeWolf) [inline CONTRIBUTING.md:14]: In the future, we could have a section for custom module here?

File: CONTRIBUTING.md
Comment (NitinAgg) [reply CONTRIBUTING.md:27]: We already have issue templates: https://github.com/run-x/opta/tree/main/.github/ISSUE_TEMPLATE 

I am just repeating things from there. Helps folks who haven't created an issue yet, on what the process would look like when they do.

File: CONTRIBUTING.md
Comment (NitinAgg) [reply CONTRIBUTING.md:14]: 💯 %

File: CONTRIBUTING.md
Comment (RemyDeWolf) [reply CONTRIBUTING.md:27]: cool, sorry I forgot to check

**Review by RemyDeWolf — APPROVED** [review]

**Pull request merged**
